content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+
+### Restrict
+
+The `Restrict` action will prevent the update or deletion of any parent records which have connected children. 
+
+> **Note**: The children records can still be deleted
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: Restrict)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `Restrict`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete the `User`, you will be able to delete (all) the posts however.
+
+<Admonition type="warning">
+
+The `Restrict` action is not is **not** available on [SQL Server](../../database-connectors/microsoft-sql-server) and will trigger a schema validation error. Instead, use the equivalent [`NoAction`](#noaction).
+
+</Admonition>
+
+### NoAction
+
+The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependant on the database being used. But in general they both have the same functionality. In Postgres however, `NoAction` defers the check until later in the transaction.
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: NoAction)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `NoAction`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete them. This will result in a runtime error.
+
+### SetNull
+
+The `SetNull` action will set relation [scalar fields](./data-model#scalar-fields) to null if the relation is updated or deleted. If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to `Null`
```

Comment:
```suggestion
If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to `NULL`. It will be left in the database, floating there without a connection to any `User`.
```